### PR TITLE
fix: fix recursion-like stack overflow error caused by the old render…

### LINKED ISSUE #3379: Version 13 , old rendering behavior, combined with a large number of execution calls, will lead to an "Maximum call stack size exceeded" error
**Marked version: 13**

**Describe the bug**

as title, code will throw "Maximum call stack size exceeded" error

**To Reproduce**

1. `npm install --save marked@13.0.2`
2. run code like below

```js
const marked = require('marked')

const content = '\n## 1. some title\n\nsimple content'

marked.use({
	renderer: {
		heading(text, level) {
			const escapedText = encodeURIComponent(text).replaceAll('%', '.');
			return `<h${level} id="${escapedText}">
									${text}
					</h${level}>`;
					
		},
	},
});

for (const _ of new Array(10000).fill(0)) {
	marked.parse(content);
}
```
4. code will throw error "RangeError: Maximum call stack size exceeded"
image

**some code dig**

As I read through the code, it seems that, in simple terms, reassignment `rendererFunc` leads to recursive-like calls (each execution results in an additional call).

image

### PR DESCRIPTION
**Pull request by yijian166**
fix #3379

### CODE REVIEW

File: src/Instance.ts [hunk 1]
```diff
@@ -158,7 +158,7 @@ export class Marked {
           renderer[rendererProp] = (...args: unknown[]) => {
             if (!pack.useNewRenderer) {
               // TODO: Remove this in next major version
-              rendererFunc = this.#convertRendererFunction(rendererFunc, rendererProp, renderer) as GenericRendererFunction;
+              rendererFunc = this.#convertRendererFunction(pack.renderer![rendererProp] as GenericRendererFunction, rendererProp, renderer) as GenericRendererFunction;
             }
```
Comment (UziTech) [inline src/Instance.ts:162]: I feel like moving this out of the `renderer[rendererProp]` function would be better. There is no reason we have to assign rendererFunc every time the function is called.

**Review by UziTech — CHANGES_REQUESTED** [review]
Nice catch!

File: src/Instance.ts (hunk 1 above)
Comment (yijian166) [reply src/Instance.ts]: yes, moving this out of make sense.

**Review by UziTech — APPROVED** [review]
Thanks for fixing this 💯

**Comment by yijian166** [comment]
@UziTech sorry, wrong click ~~

**Pull request merged**